optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/SwapMove.java

```diff
@@ -151,7 +151,7 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         } else if (o instanceof SwapMove) {
-            SwapMove other = (SwapMove) o;
+            SwapMove<?> other = (SwapMove) o;
```

Comment:
This PR is a good idea.
For this line, can't we avoid the ? and simply do:

    SwapMove<Solution_> other = (SwapMove) o;